repository_service_tuf_worker/repository.py

```diff
@@ -398,23 +401,66 @@ def bootstrap(
     ) -> Dict[str, Any]:
         """
         Bootstrap the Metadata Repository
-
-        Add the online Keys to the Key Storage backend and the Signed Metadata
-        to the Storage Backend.
         """
-        if payload.get("settings") is None:
-            raise ValueError("No settings in the payload")
-
-        metadata = payload.get("metadata")
-        if metadata is None:
-            raise (ValueError("No metadata in the payload"))
+        tuf_settings = payload.get("settings")
+        if tuf_settings is None:
+            raise KeyError("No 'settings' in the payload")
+
+        root_metadata = payload.get("metadata")
+        if root_metadata is None:
+            raise KeyError("No 'metadata' in the payload")
+
+        # Saves the `Root` metadata in the backend storage service and returns
+        # online public key (it uses the `Timestamp` key as reference)
+        root: Metadata[Root] = Metadata.from_dict(root_metadata[Root.type])
+        self._persist(root, Root.type)
+        _keyid: str = root.signed.roles[Timestamp.type].keyids[0]
+        public_key = root.signed.keys[_keyid]
+
+        # Top level roles (`Targets`, `Timestamp``, `Snapshot`) initialization
+        targets: Metadata[Targets] = Metadata(Targets())
+        snapshot: Metadata[Snapshot] = Metadata(Snapshot())
+        timestamp: Metadata[Timestamp] = Metadata(Timestamp())
+
+        # Succinct delegated roles (`bins`)
+        succinct_roles = SuccinctRoles(
+            [],
+            1,
+            int(log(tuf_settings["services"]["number_of_delegated_bins"], 2)),
```

Comment:
I think it will be clearer if you calculate it before in a value or maybe use keyword arguments.
 This could help future readers understand the `SuccinctRoles` arguments better as right now it's not immediately clear which argument it represents especially this one (the third one).